### PR TITLE
chore: prepare for TS defs generation

### LINKED ISSUE #35: Generate TypeScript definitions for P3
Child of vaadin/vaadin-core#186

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #35 

The result of running `magi p3-convert` locally (without auto-generated comments in `.d.ts`):

```
> @vaadin/vaadin-element-mixin@2.3.2 generate-typings /Users/sergey/vaadin/vaadin-element-mixin
> gen-typescript-declarations --outDir . --verify

Loading config from "gen-tsd.json".
Writing type declarations to /Users/sergey/vaadin/vaadin-element-mixin
Verifying type declarations
Compilation successful
```

## vaadin-dir-helper.d.ts

```ts
/**
 * Helper that provides a set of functions for "dir="rtl".
 */
declare class DirHelper {

  /**
   * Get the scroll type in the current browser view.
   *
   * @returns the scroll type. Possible values are `default|reverse|negative`
   */
  static detectScrollType(): string;

  /**
   * Get the scrollLeft value of the element relative to the direction
   *
   * @param scrollType type of the scroll detected with `detectScrollType`
   * @param direction current direction of the element
   * @returns the scrollLeft value.
   */
  static getNormalizedScrollLeft(scrollType: string, direction: string, element: Element|null): number;

  /**
   * Set the scrollLeft value of the element relative to the direction
   *
   * @param scrollType type of the scroll detected with `detectScrollType`
   * @param direction current direction of the element
   * @param scrollLeft the scrollLeft value to be set
   */
  static setNormalizedScrollLeft(scrollType: string, direction: string, element: Element|null, scrollLeft: number): void;
}

export {DirHelper};
```

## vaadin-dir-mixin.d.ts

```ts
import {DirHelper} from './vaadin-dir-helper.js';

export {DirMixin};

declare function DirMixin<T extends new (...args: any[]) => {}>(base: T): T & DirMixinConstructor;

interface DirMixinConstructor {
  new(...args: any[]): DirMixin;
}

export {DirMixinConstructor};

interface DirMixin {
  readonly dir: string|null|undefined;
  connectedCallback(): void;
  attributeChangedCallback(name: any, oldValue: any, newValue: any): void;
  disconnectedCallback(): void;
  __getNormalizedScrollLeft(element: Element|null): string;
  __setNormalizedScrollLeft(element: Element|null, scrollLeft: number): any;
}
```

## vaadin-element-mixin.d.ts

```ts
/// <reference path="custom_typings/vaadin-usage-statistics.d.ts" />

import {idlePeriod} from '@polymer/polymer/lib/utils/async.js';

import {Debouncer} from '@polymer/polymer/lib/utils/debounce.js';

import {enqueueDebouncer} from '@polymer/polymer/lib/utils/flush.js';

import {DirMixin} from './vaadin-dir-mixin.js';

import {usageStatistics} from '@vaadin/vaadin-usage-statistics/vaadin-usage-statistics.js';

export {ElementMixin};

declare function ElementMixin<T extends new (...args: any[]) => {}>(base: T): T & ElementMixinConstructor & DirMixinConstructor;

import {DirMixinConstructor} from './vaadin-dir-mixin.js';

interface ElementMixinConstructor {
  new(...args: any[]): ElementMixin;
  finalize(): void;
}

export {ElementMixinConstructor};

interface ElementMixin extends DirMixin {
}
```

### CODE REVIEW

File: vaadin-dir-mixin.html [hunk 1]
```diff
@@ -93,10 +97,20 @@
       }
     }
 
+    /**
+     * @param {Element} element
+     * @return {string}
```
Comment (tomivirkki) [inline vaadin-dir-mixin.html]: Should be `{number}`

**Review by tomivirkki — CHANGES_REQUESTED** [review]

File: vaadin-dir-mixin.html (hunk 1 above)
Comment (web-padawan) [reply vaadin-dir-mixin.html]: Done.

File: custom_typings/vaadin.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+interface Vaadin {
+  developmentModeCallback: {
+    'usage-statistics': () => void;
+    'vaadin-license-checker'?: () => void;
+  };
+  registrations: Array<{ is: string }>;
```
Comment (tomivirkki) [inline custom_typings/vaadin.d.ts]: These should be optional. Currently getting this:
Screenshot 2020-04-16 11 54 32

Comment (web-padawan) [reply custom_typings/vaadin.d.ts]: Done.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**